### PR TITLE
Added the `VSCode-SessionId` for extension queries Issue2961

### PR DESCRIPTION
**Pull request by prashantvc**
Added the `VSCode-SessionId` for extension queries

### CODE REVIEW

**Review by sandy081 — CHANGES_REQUESTED** [review]

File: src/vs/platform/extensionManagement/common/extensionGalleryService.ts [hunk 1]
```diff
@@ -923,12 +923,14 @@ abstract class AbstractExtensionGalleryService implements IExtensionGalleryServi
...
 		const data = JSON.stringify(query.raw);
+		const { sessionId } = await this.telemetryService.getTelemetryInfo();
 		const headers = {
 			...commonHeaders,
 			'Content-Type': 'application/json',
 			'Accept': 'application/json;api-version=3.0-preview.1',
 			'Accept-Encoding': 'gzip',
-			'Content-Length': String(data.length)
+			'Content-Length': String(data.length),
+			'VSCode-SessionId': sessionId
```
[the remark this answers is not in the document]
Comment (prashantvc) [reply src/vs/platform/extensionManagement/common/extensionGalleryService.ts]: I have submitted new commit, could you take a look?

File: src/vs/platform/extensionManagement/test/common/extensionGalleryService.test.ts [hunk 2]
```diff
@@ -153,3 +158,51 @@ suite('Extension Gallery Service', () => {
 		return { version, targetPlatform } as IRawGalleryExtensionVersion;
 	}
 });
+
+class MockTelemetryService implements ITelemetryService {
```
Comment (sandy081) [inline src/vs/platform/extensionManagement/test/common/extensionGalleryService.test.ts:162]: Please use the NullTelemetryService instead

https://github.com/microsoft/vscode/blob/ade80ca128741e2aa97e836a1ff0ad7cf8d67fa7/src/vs/platform/telemetry/common/telemetryUtils.ts#L46

**Review by sandy081 — CHANGES_REQUESTED** [review]

File: src/vs/platform/extensionManagement/test/common/extensionGalleryService.test.ts (hunk 2 above)
Comment (prashantvc) [reply src/vs/platform/extensionManagement/test/common/extensionGalleryService.test.ts]: Thanks, fixed it!
What is the best way to send PR, should I talk to you first?
I am still getting familiar with the code base; I don't want to be wasting your time in review :)

Comment (sandy081) [reply src/vs/platform/extensionManagement/test/common/extensionGalleryService.test.ts]: Either way works. If it is a small one like this, you can go ahead and create a PR like here and ask for review. If its a big change reaching out and discussing will definitely save time for you 👍

**Review by sandy081 — APPROVED** [review]

**Pull request merged**